tests/test_templates.py

```diff
@@ -87,30 +88,31 @@
 # Parameters shared between test classes
 
 # qubit templates, constant args and kwargs for 2 wires
-QUBIT_CONSTANT_INPUT = [(StronglyEntanglingLayers, [[[[4.54, 4.79, 2.98], [4.93, 4.11, 5.58]],
-                                                     [[6.08, 5.94, 0.05], [2.44, 5.07, 0.95]]]], {}),
-                        (RandomLayers, [[[0.56, 5.14], [2.21, 4.27]]], {}),
-                        (AngleEmbedding, [[1., 2.]], {}),
-                        (QAOAEmbedding, [[1., 2.], [[0.1, 0.1, 0.1]]], {})
-                        ]
+QUBIT_PARAMS_KWARGS = [(StronglyEntanglingLayers, [[[[4.54, 4.79, 2.98], [4.93, 4.11, 5.58]],
+                                                    [[6.08, 5.94, 0.05], [2.44, 5.07, 0.95]]]], {}),
+                       (RandomLayers, [[[0.56, 5.14], [2.21, 4.27]]], {}),
+                       (AngleEmbedding, [[1., 2.]], {}),
+                       (QAOAEmbedding, [[1., 2.], [[0.1, 0.1, 0.1]]], {}),
+                       (Single, [], {'unitary': qml.RX, 'wires': [0, 1], 'parameters': [[1.], [1.]]})
+                       ]
```

Comment:
There is one problem here: The integration tests pass the second object in the (*, *, *) entries as positional and keyword arguments to the qnode to check that both works. But in ``Single``,  the differentiable ``parameters`` is a kwarg, while the non-differentiable ``unitary`` is an arg.

That exposes a weakness of the tests.

I propose doing a separate test refactor, rather than fixing it in this PR.

Reply:
Would making each test more explicit, with their own test data/avoiding parametrizing, make it more maintainable?

I tend to be of the opinion that parametric test data is best for parametrizing test data for specific unit tests, rather than also parametrizing the functions to be tested.

Reply:
The nice thing here is that we just need to add a new line for a new template, and it gets tested with all interfaces and when calling it before/after any other template...